### DISCUSSION TITLE
How to get a Full Tailwindcss3.css, or at least "rich enough" for testing purposes ?

### DISCUSSION THREAD

**Discussion by Stephano120** [tailwindlabs/tailwindcss#11063] [Help] [answered]
Hi there,

I used tailwind 2 back in 2019 with a Dash App. It was great, as far as I remember it was possible to use a kind of tailwind2.full.css, put it in the dash app folder, and start playing around with all the classes/utilities. For dev/testing purposes, it was perfect.

Today I'm trying to use tailwind 3, with a new dash app. Problem, it is based on this "JIT" concept, which requires to have an input.css and a tailwind.config.js before generating an output.css with ./tailwindcss-windows-x64.exe -i input.css -o output.css --watch

This works well if the user knows what class/utilities will be used _in the future_, or alternatively if he already started to use these classes in his project, blindly (because without the CSS file at the right place, the browser will not render any CSS). 
However if the starting point is zero, then the output.css just has 517 rows.

My dev station does not have access to the internet. I can't use the cdn.

I did search for similar discussions; I found 
https://github.com/tailwindlabs/tailwindcss/discussions/6347
https://github.com/tailwindlabs/tailwindcss/discussions/6347#discussioncomment-2185719
https://github.com/tailwindlabs/tailwindcss/discussions/6347#discussioncomment-1905577
https://github.com/tailwindlabs/tailwindcss/discussions/6256#discussioncomment-1747715

I did try the safelisting, but I always get a 10Ko output.css file with 517 lines. Since my input.css file is almost empty, when I make the output.css it warns that "No utility classes were detected in your source files."
The input.css contains 
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```
What do I have to change to get a decent-sized-and-rich-enough CSS file ? Everything is executed on my local machine, I don't really care if the final output.CSS is 50MB. I can purge it at the end, later.

KR

**Comment by wongjn** [comment, marked as answer]
Try configuring `safelist` correctly, like:

```js
module.exports = {
  safelist: [
    {
      pattern: /^/,
      variants: ['hover', 'focus', 'focus-within', …],
    },
  ],
};
```

**Reply by Stephano120** [reply]
Thanks ! I now get a nice 46MB file. Much better for testing :-)
